x/supplier/module/tx_stake_supplier.go

```diff
@@ -41,13 +41,28 @@ $ poktrolld tx supplier stake-supplier --config stake_config.yaml --keyring-back
 				return err
 			}
 
+			// Ensure the --from flag is set before getting the client context.
+			if cmd.Flag(flags.FlagFrom) == nil {
+				cmd.Flags().Set(flags.FlagFrom, supplierStakeConfigs.OwnerAddress)
+			}
+
 			clientCtx, err := client.GetClientTxContext(cmd)
 			if err != nil {
 				return err
 			}
 
+			// Ensure the from address is the same as the owner address in the stake config file.
+			if clientCtx.GetFromAddress().String() != supplierStakeConfigs.OwnerAddress {
```

Comment:
Thoughts on moving this into `ValidateBasic`?

Reply:
We can wrap this check in a `EnsureOwner` but I don't think `ValidateBasic` is the best fit, as it is usually about intrinsic validation of a type; while `clientCtx.GetFromAddress().String()` is an external value we want to validate against.

Reply:
Let's do it!